gdk4/src/subclass/content_provider.rs

```diff
@@ -0,0 +1,445 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+use crate::{Clipboard, ContentFormats, ContentProvider};
+
+use crate::subclass::prelude::*;
+use glib::object::IsA;
+use glib::translate::*;
+use glib::{Cast, Object, ToValue, Value};
+use std::future::Future;
+use std::pin::Pin;
+
+pub trait ContentProviderImpl: ContentProviderImplExt + ObjectImpl {
+    fn content_changed(&self, provider: &Self::Type) {
+        self.parent_content_changed(provider)
+    }
+
+    fn attach_clipboard(&self, provider: &Self::Type, clipboard: &Clipboard) {
+        self.parent_attach_clipboard(provider, clipboard)
+    }
+
+    fn detach_clipboard(&self, provider: &Self::Type, clipboard: &Clipboard) {
+        self.parent_detach_clipboard(provider, clipboard)
+    }
+
+    fn formats(&self, provider: &Self::Type) -> ContentFormats {
+        self.parent_formats(provider)
+    }
+
+    fn storable_formats(&self, provider: &Self::Type) -> ContentFormats {
+        self.parent_storable_formats(provider)
+    }
+
+    fn write_mime_type_future<P: IsA<gio::OutputStream> + Clone + 'static>(
+        &self,
+        provider: &Self::Type,
+        mime_type: &str,
+        stream: &P,
```

Comment:
```suggestion
        stream: &gio::OutputStream,
```

and remove `P` in all these functions. As this is passed in from C code, you will never ever have a different type here and having a concrete type is easier for user code implementing this.

Reply:
make sense